### PR TITLE
CoordinateService: add getLength/getArea methods

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
resolves partially #2189 

- [x] coordinateService.getLength()
- [x] coordinateService.getArea() 
- [x] usage in routeStats.provider
- [x] usage in elevation.provider

### CODE REVIEW

File: src/services/OlCoordinateService.js [hunk 1]
```diff
@@ -212,4 +214,61 @@ export class OlCoordinateService {
...
+					return sum;
+				}
+				const next = coordinates[index + 1];
+				const r = wgs84.Inverse(current[1], current[0], next[1], next[0]);
+
+				return sum + r.s12;
+			}, 0);
+		};
+		const wgs84Coordinates =
+			coordinateRepresentation?.code && coordinateRepresentation?.code !== 4326
```
Comment (taulinger) [inline src/services/OlCoordinateService.js:239]: `coordinateRepresentation?.code && coordinateRepresentation?.code !== 4326`

- unneeded check for code 4326 (ol.transform can safely always be called even if the coordinates are already in the target SRID)
- I would suggest throwing an error when `CoordinateRepresentation` contains no `srid` property

Comment (thiloSchlemmer) [reply src/services/OlCoordinateService.js:239]: ok, so....

- coordinateRepresentation is not optional
- and something like `if(!coordinateRepresentation.code) throw new Error('The coordinates must have a reprojectable CoordinateRepresentation (code!== null)')`

**Review by taulinger — COMMENTED** [review]
grafik

**Review by taulinger — COMMENTED** [review]
grafik

**Review by taulinger — APPROVED** [review]
Yeah, thx

**Pull request merged**